routes/login.js

```diff
@@ -0,0 +1,142 @@
+                                              "Internal error occurred!"
+                                          )
+                                      );
+                        } else {
+                            res.status(401).send(
+                                new Error("UNAUTHORIZED", "Password incorrect!")
+                            );
+                        }
+                    }
+                });
+            })
+            .catch(() => {
+                res.status(400).send(
+                    new Error("BAD_REQUEST", "User not found!")
+                );
+            });
+    }
+});
+
+router.post("/add", (req, res) => {
+    const { zprn, firstName, lastName, password, department, role } = req.body;
+    const { er, value } = validateData(
+        zprn,
+        firstName,
+        lastName,
+        password,
+        department,
+        role
+    );
+    if (er != undefined) {
+        res.status(400).send(
+            new Error("BAD_REQUEST", error.details[0].message)
+        );
+    } else {
+        bcrypt.genSalt(10, (e, salt) => {
+            if (!e) {
+                bcrypt.hash(password, salt, (error, hash) => {
+                    if (error) {
+                        res.status(500).send(
+                            new Error("INTERNAL_SERVER_ERROR", error)
+                        );
+                    } else {
+                        if (!hash) {
+                            res.status(500).send(
+                                new Error(
+                                    "INTERNAL_SERVER_ERROR",
+                                    "An internal error occured!"
+                                )
+                            );
+                        } else {
+                            userService
+                                .addUser(
+                                    zprn,
+                                    firstName,
+                                    lastName,
+                                    hash,
+                                    department,
+                                    role
+                                )
+                                .then(() => {
+                                    res.status(20).json({
+                                        code: "CREATED",
+                                        msg: "User added",
+                                    });
+                                })
+                                .catch((err) => {
+                                    res.status(500).send(
+                                        new Error("INTERNAL_SERVER_ERROR", err)
+                                    );
+                                });
+                        }
+                    }
+                });
+            }
+            res.status(500).send(new Error("INTERNAL_SERVER_ERROR", e));
+        });
+    }
+});
+
+function validateCredentials(zprn, password) {
+    const schema = Joi.object({
```

Comment:
`zprn` is of specific characters size. Also we should add similar validations to other parameters as well. The max size and min size should be specified here.

Reply:
Will implement this!